### DISCUSSION TITLE
Support pdf printable version of test reports

### DISCUSSION THREAD

**Discussion by MarcoMinghini** [INSPIRE-MIF/helpdesk-validator#784] [Ideas]
Currently the only format to download a test report is .html, which is not a printer-friendly format.

A possible improvement is to add the option to download the test report in pdf. 
Eventually, before the download, a pop-up window can be displayed to select the level of detail of the pdf to be generated, e.g. as shown in this mockup:

![imagen](https://user-images.githubusercontent.com/25343632/46080982-a2c96780-c19b-11e8-917c-3570db4b8856.png)

The download button for html and pdf download will be shown in both the report list (when opening the details of a report) and in the page of the report itself.

Do you think this is a useful improvement to be added? What is the level of detail you would suggest for the pdf (summary, full detail, user-defined detail with selectable options, etc.)?

**Comment by MarieLambois** [comment]
I think it would be useful only for a summary view, not the full technical report.

**Comment by thijsbrentjens** [comment]
If we need to choose, I agree with @MarieLambois . I think it will be useful to have a list of the tests that failed and passed, the resource tested and date of the test, but don't include the technical details  / error messages itself. A URL to the online report could point the user to the technical details and logging.

But if it is not too hard, let the user choose the level of detail as @MarcoMinghini proposes, would be nice. I can't recall users that requested this, but maybe there are cases where someone likes it to have.

**Comment by MarcoMinghini** [comment]
[2017.4 meeting 2018-11-15] the improvement proposal is considered to be useful and nice to have. The level of detail of the report (or the possibility for the user to select/change the level of detail) is still to be agreed - some are in favour of a summary view, some of a fully detailed report.

**Comment by alitka** [comment]
Implementation or realisation of a printable version of test reports (PDF-file) in the german GDI-DE Testsuite: 

![grafik](https://user-images.githubusercontent.com/29542183/51116241-da235900-180a-11e9-99d4-9b55c450c0ee.png)

Example of a test-report: [pdfReport_GDI-DE_Testsuite.pdf](https://github.com/inspire-eu-validation/community/files/2755345/pdfReport_GDI-DE_Testsuite.pdf)

**Comment by AntoRot** [comment]
I agree to let the user to select the level of detail for the report as @MarcoMinghini proposed.

**Comment by IGNCNIG** [comment]
It would be nice if the user could select the level of detail for the report .
